src/mimemail.erl

```diff
@@ -1145,8 +1153,9 @@ dkim_hash_data(CanonicHeaders, DkimHeader) ->
 dkim_sign(DataHash, {pem_plain, PrivBin}) ->
 	[PrivEntry] = public_key:pem_decode(PrivBin),
 	Key = public_key:pem_entry_decode(PrivEntry),
-	public_key:sign({digest, DataHash}, sha256, Key);
+	public_key:sign({digest, DataHash}, sha512, Key);
 dkim_sign(DataHash, {pem_encrypted, EncPrivBin, Passwd}) ->
+    %% TODO: Ed25519
 	[EncPrivEntry] = public_key:pem_decode(EncPrivBin),
 	Key = public_key:pem_entry_decode(EncPrivEntry, Passwd),
 	public_key:sign({digest, DataHash}, sha256, Key).
```

Comment:
So, given we hit a known bug in OTP and that fix is on the way, we probably can backport the fix so it works on a broader set of OTP versions?

So, smth like this

```erlang
-include_lib("public_key/include/public_key.hrl").
.....

dkim_sign(DataHash, Algorithm, PrivateKey) ->
  Key = parse_key(PrivateKey, Algorithm),
  Digest = case Algorithm of
      'rsa-sha256' -> sha256;
      'ed25519-sha256' -> none
  end,
  public_key:sign({digest, DataHash}, Digest, Key).

parse_key({pem_plain, PrivBin}, Algo) ->
    [PrivEntry] = public_key:pem_decode(PrivBin),
   fix_otp5156(public_key:pem_entry_decode(PrivEntry), Algo);
parse_key({pem_encrypted, EncPrivBin, Passwd}, Algo) ->
   [EncPrivEntry] = public_key:pem_decode(EncPrivBin),
	fix_otp5156(public_key:pem_entry_decode(EncPrivEntry, Passwd)).

fix_otp5156(#'PrivateKeyInfo'{privateKey = PK} = Key, 'ed25519-sha256') when byte_size(PK) =:= 34 ->
  % backport of https://github.com/erlang/otp/pull/5157
  PK1 = public_key:der_decode('CurvePrivateKey', PK),
  Key#'PrivateKeyInfo'{privateKey = PK1};
fix_otp5156(Key, _) ->
  Key.
```

Reply:
Wow! Thanks for digging that deep.
If it is easy to backport, also for future maintenance, then yes.

Otherwise I would be ok if it is OTP 24.1+.
Anyway I think not too much people would bother about DKIM with Ed25519.

Reply:
Yep, it seems the easiest way would be to only allow it on OTP-24.1+. See the discussion in https://github.com/erlang/otp/pull/5157#issuecomment-912409815

Reply:
I totally agree.